### PR TITLE
Kernel+ProfileViewer: Monitor and display more filesystem events

### PR DESCRIPTION
**Pull request by jakubberkop**
This pull request adds monitoring for filesystem related syscalls - open, close, read, readv and pread.
To enable profiling of those syscalls, call `profile -e -a -t filesystem", and open new profile using Profiler.

Example run of the filesystem profiling:
image

To facilitate displaying new information, "Filesystem events" tab was added to the Profiler. Filesystem events are accumulated based on their path. Cumulative duration and event count is show for every file and folder.

Monitoring of filesystem events during boot was also tested, but it makes the startup time for the system about 10x slower (currently this requires the Profiler, so that it accepts unknown event types - kfree, kmalloc, syscall, context_switch, etc.)

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
Sorry this has been waiting for a review so long! You've got some conflicts now, because of a change to error propagation. Please fix those.

I'm not a kernel guy but this is a cool feature and seems to work well, so it'd be nice to get it merged. :^)

File: Kernel/Tasks/PerformanceEventBuffer.h [hunk 1]
```diff
@@ -145,7 +206,7 @@ class PerformanceEventBuffer {
     size_t m_count { 0 };
     NonnullOwnPtr<KBuffer> m_buffer;
 
-    HashMap<NonnullOwnPtr<KString>, size_t> m_strings;
+    SpinlockProtected<HashMap<NonnullOwnPtr<KString>, size_t>, LockRank::None> m_strings;
```
Comment (AtkinsSJ) [inline Kernel/Tasks/PerformanceEventBuffer.h:209]: Adding a lock around this is a significant change on its own. Can you split that into a separate commit?

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]

File: Kernel/Tasks/PerformanceEventBuffer.h (hunk 1 above)
Comment (jakubberkop) [reply Kernel/Tasks/PerformanceEventBuffer.h:209]: I have added a separate [commit](https://github.com/SerenityOS/serenity/pull/19950/commits/904bd1a1609c0572974678afd10562ec2b948143) as requested.
